File: examples/python/11.poo/06.static/02-static_method.py
#!/usr/bin/env python3
# https://pynative.com/python-static-method/

class Person:
    rol = "jedi"

    def __init__(self, name='unkown'):
        self.name = name

    @staticmethod
    def class_method():
        print('Inside class method: rol=')

    def instance_method(self):
        print('Inside instance method: name=' + self.name + ", rol=" + self.rol)

print('=== call static method ===')
Person.class_method()
Person().instance_method()

print('=== can be called using object ===')
yoda = Person("Yoda")
yoda.class_method()
yoda.instance_method()

File: examples/2.high-level/python/11.poo/06.static/02-static_method.rb
#!/usr/bin/env ruby

class Person
  def initialize(name)
    @name = name
  end

  def self.class_method
    puts "Inside class method"
  end

  def instance_method
    puts "Inside instance method " + @name
  end
end

# call static method
Person.class_method

# can't  be called using object
yoda = Person.new("Yoda")
yoda.instance_method
begin
  yoda.class_method
rescue NoMethodError
  puts "NoMethodError: static methods belongs to Class not to instances"
end
